src/actions/datarobot/datarobot.ts

```diff
@@ -0,0 +1,120 @@
+import * as Hub from "../../hub"
+
+import * as httpRequest from "request-promise-native"
+
+const DR_API_URL = "https://app.datarobot.com/api/v2"
+
+export class DataRobotAction extends Hub.Action {
+
+  name = "datarobot"
+  label = "DataRobot - Create New Project"
+  iconName = "datarobot/dr-head.svg"
+  description = "Send data to DataRobot and create a new project."
+  supportedActionTypes = [Hub.ActionType.Query, Hub.ActionType.Dashboard]
+  supportedFormats = [Hub.ActionFormat.Csv]
+  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+  supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+  requiredFields = []
+  usesStreaming = true
+  params = [
+    {
+      name: "datarobot_api_token",
+      label: "DataRobot API Token",
+      description: "API Token from https://app.datarobot.com/account/me",
+      required: true,
+      sensitive: true,
+    },
+  ]
+
+  async execute(request: Hub.ActionRequest) {
+    const options = this.getRequestOptions(request)
+    try {
+      await httpRequest.post(options).promise()
+      return new Hub.ActionResponse({ success: true })
+    } catch (e) {
+      return new Hub.ActionResponse({ success: false, message: e.message })
+    }
+  }
+
+  async form(request: Hub.ActionRequest) {
+    const form = new Hub.ActionForm()
+
+    if (!request.params.datarobot_api_token) {
+      form.error = "No DataRobot API Token configured; consult your Looker admin."
+      return form
+    }
+
+    try {
+      await this.validateDataRobotToken(request.params.datarobot_api_token)
+
+      form.fields = [
+        {
+          label: "The name of the project to be created",
+          name: "projectName",
+          required: false,
+          type: "string",
+        },
+      ]
+    } catch (e) {
+      form.error = this.prettyDataRobotError(e)
+    }
+
+    return form
+  }
+
+  private getRequestOptions(request: Hub.ActionRequest) {
+    const options = {
+      url: `${DR_API_URL}/projects/`,
+      headers: {
+        Authorization: `Token ${request.params.datarobot_api_token}`,
+      },
+    }
+
+    const url = request.scheduledPlan && request.scheduledPlan.downloadUrl
```

Comment:
Update: I've fixed item 2 on the Looker end, which will make it into 5.24, so I'll update this PR to specify this as the minimum Looker version that this action supports.

Reply:
Amazing news:) Excited to get this merged.

Reply:
Yes, basically we'll add `minimumSupportedLookerVersion = "5.24.0"` to this action, which will make it only show up on or after 5.24. Looker 5.24 and later will properly guarantee that it will only send actions in a supported download setting, so yes you'll be able to rely on Looker only sending you the data via URL. It's worth noting that there is a 5 minute time limit to initiate the request via the download URL.